### ISSUE TITLE
Wrong links in searches when using `base`

### ISSUE THREAD

**Issue by JuanM04** [vuejs/vitepress#2221]
### Describe the bug

When setting some `base`, the links given by the new local search (by @Akryum) are wrong.

### Reproduction

https://stackblitz.com/edit/vite-p7s8di?file=docs/.vitepress/config.ts

### Expected behavior

For instance, if I have a base `/docs/` and I want to go to `example.md`. If I search for "example", instead of going to `/docs/example`, it goes to `/docs/docs/example`

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 AMD Ryzen 5 3600XT 6-Core Processor
    Memory: 16.93 GB / 31.93 GB
  Binaries:
    Node: 18.15.0 - ~\AppData\Local\fnm_multishells\19660_1681579277222\node.EXE
    npm: 9.5.0 - ~\AppData\Local\fnm_multishells\19660_1681579277222\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1555.0), Chromium (112.0.1722.46)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    vitepress: 1.0.0-alpha.66 => 1.0.0-alpha.66
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by Akryum** [comment]
@brc-dd could it be because of 
https://github.com/vuejs/vitepress/commit/609d447ab50b2b8fb78a174e7d5aa0ff52411b0e ?

**Comment by brc-dd** [comment]
Yeah working on it. Actually if base is not set only then we need to add `/`.

**Comment by brc-dd** [comment]
fixed by 25a1fe9
